### PR TITLE
Normalize display string % encoding to lowercase.

### LINKED ISSUE #2618: Require Display Strings % encoding to be serialised in lowercase
It's annoying to test for otherwise. Still allow both cases in parse.

### PR DESCRIPTION
**Pull request by mnot**
Fixes #2618.

### CODE REVIEW

**Comment by bsdphk** [comment]
Can I suggest we use 'converted' instead of 'normalized' ?

**Comment by mnot** [comment]
WFM, will update.

File: draft-ietf-httpbis-sfbis.md [hunk 1]
```diff
@@ -711,7 +711,7 @@ Given a sequence of Unicode codepoints as input_sequence, return an ASCII string
 3. For each byte in byte_array:
    1. If byte is %x25 ("%"), %x22 (DQUOTE), or in the ranges %x00-1f or %x7f-ff:
       1. Append "%" to encoded_string.
-      2. Let encoded_byte be the result of applying base16 encoding ({{Section 8 of !RFC4648}}) to byte.
+      2. Let encoded_byte be the result of applying base16 encoding ({{Section 8 of !RFC4648}}) to byte, with any alphabetic characters normalized to lowercase.
```
Comment (mnot) [inline draft-ietf-httpbis-sfbis.md:714]:
```suggestion
      2. Let encoded_byte be the result of applying base16 encoding ({{Section 8 of !RFC4648}}) to byte, with any alphabetic characters converted to lowercase.
```

File: draft-ietf-httpbis-sfbis.md [hunk 2]
```diff
@@ -993,7 +993,8 @@ Given an ASCII string as input_string, return a sequence of Unicode codepoints.
    2. If char is in the range %x00-1f or %x7f-ff (i.e., it is not in VCHAR or SP), fail parsing.
    3. If char is "%":
       1. Let octet_hex be the result of consuming two characters from input_string. If there are not two characters, fail parsing.
-      2. Let octet be the result of hex decoding octet_hex ({{Section 8 of !RFC4648}}), in a case-insensitive fashion. If decoding fails, fail parsing.
+      2. If octet_hex contains characters outside the range %x30-39 or %x61-66 (i.e., it is not in 0-9 or lowercase a-f), fail parsing.
+      2. Let octet be the result of hex decoding octet_hex ({{Section 8 of !RFC4648}}).
```
Comment (martinthomson) [inline draft-ietf-httpbis-sfbis.md:997]: Two “2. “ bullets?

**Pull request merged**